core/arch/x86/emit_utils.c

```diff
@@ -1354,13 +1354,59 @@ append_restore_simd_reg(dcontext_t *dcontext, instrlist_t *ilist, bool absolute)
         int i;
         uint opcode = move_mm_reg_opcode(true /*align16*/, true /*align32*/);
         ASSERT(proc_has_feature(FEATURE_SSE));
-        for (i = 0; i < proc_num_simd_saved(); i++) {
+        instr_t *post_restore = NULL;
+        instr_t *pre_avx512_restore = NULL;
+        if (ZMM_ENABLED()) {
+            post_restore = INSTR_CREATE_label(dcontext);
+            pre_avx512_restore = INSTR_CREATE_label(dcontext);
+            APP(ilist,
+                INSTR_CREATE_mov_imm(
+                    dcontext, opnd_create_reg(SCRATCH_REG1),
+                    OPND_CREATE_INTPTR((ptr_int_t)&d_r_avx512_code_in_use)));
+            APP(ilist,
+                XINST_CREATE_load_1byte_zext4(dcontext, opnd_create_reg(SCRATCH_REG1),
+                                              OPND_CREATE_MEM8(SCRATCH_REG1, 0)));
+            APP(ilist,
+                INSTR_CREATE_test(dcontext, opnd_create_reg(SCRATCH_REG1),
+                                  opnd_create_reg(SCRATCH_REG1)));
+            APP(ilist,
+                INSTR_CREATE_jcc(dcontext, OP_jnz,
```

Comment:
Please update the big comment in emit_utils_shared.c:1943 which lays out the cxt sw in one place

Reply:
Done.